controllers/ccruntime_controller.go

```diff
@@ -207,90 +206,107 @@ func (r *CcRuntimeReconciler) processCcRuntimeDeleteRequest() (ctrl.Result, erro
-				time.Sleep(time.Second * 60)
-
-				result, err = handlePostUninstall(r)
-				if !result.Requeue {
-					controllerutil.RemoveFinalizer(r.ccRuntime, RuntimeConfigFinalizer)
-					result, err = r.updateCcRuntime()
-					if err != nil {
-						return result, err
-					}
-					result, err = r.deleteUninstallDaemonsets()
-					prepostLabels := map[string]string{}
-					if r.ccRuntime.Spec.Config.PreInstall.Image != "" {
-						prepostLabels[PreInstallDoneLabel[0]] = PreInstallDoneLabel[1]
-					}
-					if r.ccRuntime.Spec.Config.PostUninstall.Image != "" {
-						prepostLabels[PostUninstallDoneLabel[0]] = PostUninstallDoneLabel[1]
+				result, err = r.updateCcRuntime()
+				if err != nil {
+					return result, err
+				}
 
-					}
-					err, nodes := r.getNodesWithLabels(prepostLabels)
-					if err != nil {
-						r.Log.Error(err, "an error occured when getting the list of nodes from which we want to"+
-							"remove preinstall/postuninstall labels")
-						return ctrl.Result{}, err
-					}
-					postuninstalledNodes := len(nodes.Items)
-					if !allNodesDone(postuninstalledNodes, r) {
-						return ctrl.Result{Requeue: true}, nil
-					} else {
-						if postuninstalledNodes > 0 {
-							result, err = r.removeNodeLabels(nodes)
-							if err != nil {
-								r.Log.Error(err, "removing the labels from nodes failed")
-								return ctrl.Result{}, err
-							}
+				result, err = r.deleteUninstallDaemonsets()
+				prepostLabels := map[string]string{}
+
+				if r.ccRuntime.Spec.Config.PreInstall.Image != "" {
+					prepostLabels[PreInstallDoneLabel[0]] = PreInstallDoneLabel[1]
+				}
+
+				if r.ccRuntime.Spec.Config.PostUninstall.Image != "" {
+					prepostLabels[PostUninstallDoneLabel[0]] = PostUninstallDoneLabel[1]
+
+				}
+
+				err, nodes := r.getNodesWithLabels(prepostLabels)
+				if err != nil {
+					r.Log.Error(err, "an error occured when getting the list of nodes from which we want to"+
+						"remove preinstall/postuninstall labels")
+					return ctrl.Result{}, err
+				}
+
+				postuninstalledNodes := len(nodes.Items)
+				if !allNodesDone(postuninstalledNodes, r) {
+					return ctrl.Result{Requeue: true}, nil
+				} else {
+					if postuninstalledNodes > 0 {
+						result, err = r.removeNodeLabels(nodes)
+						if err != nil {
+							r.Log.Error(err, "removing the labels from nodes failed")
+							return ctrl.Result{}, err
 						}
 					}
 				}
 			}
-			if err != nil {
-				return result, err
-			}
-			result, err = r.updateCcRuntime()
-			return result, err
 		}
 
-		result, err = r.updateUninstallationStatus(finishedNodes)
+		controllerutil.RemoveFinalizer(r.ccRuntime, RuntimeConfigFinalizer)
+
 		if err != nil {
-			r.Log.Info("Error from updateUninstallationStatus")
 			return result, err
 		}
-		result.Requeue = true
-		result.RequeueAfter = time.Second * 10
+		result, err = r.updateCcRuntime()
+		return result, err
+	}
+
+	result, err = r.setCleanupNodeLabels()
```

Comment:
this `setCleanupNodesLabels` used to only be called when `!allNodesDone(finishedNodes, r)` but now it's called always (when there are finalizers). Is it expected?

Reply:
ditto.